Frameworks/MetalPetal/MTIPixelFormat.m

```diff
@@ -17,7 +17,7 @@
 MTLPixelFormat const MTIPixelFormatYCBCR10_420_2P_sRGB = 525;
 
 BOOL MTIDeviceSupportsYCBCRPixelFormat(id<MTLDevice> device) {
-    #if TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR && !TARGET_OS_MACCATALYST
+    #if TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR && !TARGET_OS_MACCATALYST && !TARGET_OS_TV
```

Comment:
I think we should test the support of `MTLFeatureSet_tvOS_GPUFamily2_v1` here for tvOS.

Reply:
@YuAo I've run it on an Apple TV 4k and for the green values I've got 15 instead of 0. (On the device MTLFeatureSet_tvOS_GPUFamily2_v1 is supported.)
Here's the output of the pixels array:
```
  - 0 : 255
  - 1 : 15
  - 2 : 0
  - 3 : 255
  - 4 : 255
  - 5 : 15
  - 6 : 0
  - 7 : 255
  - 8 : 255
  - 9 : 15
  - 10 : 0
  - 11 : 255
  - 12 : 255
  - 13 : 15
  - 14 : 0
  - 15 : 255
  - 16 : 255
  - 17 : 15
  - 18 : 0
  - 19 : 255
  - 20 : 255
  - 21 : 15
  - 22 : 0
  - 23 : 255
  - 24 : 255
  - 25 : 15
  - 26 : 0
  - 27 : 255
  - 28 : 255
  - 29 : 15
  - 30 : 0
  - 31 : 255
  - 32 : 255
  - 33 : 15
  - 34 : 0
  - 35 : 255
  - 36 : 255
  - 37 : 15
  - 38 : 0
  - 39 : 255
  - 40 : 255
  - 41 : 15
  - 42 : 0
  - 43 : 255
  - 44 : 255
  - 45 : 15
  - 46 : 0
  - 47 : 255
  - 48 : 255
  - 49 : 15
  - 50 : 0
  - 51 : 255
  - 52 : 255
  - 53 : 15
  - 54 : 0
  - 55 : 255
  - 56 : 255
  - 57 : 15
  - 58 : 0
  - 59 : 255
  - 60 : 255
  - 61 : 15
  - 62 : 0
  - 63 : 255
```

Reply:
Nice catch! There are two problems:

1. We are not explicitly specifying any color space here.

2. We may lose some bits, when converting between YCbCr and RGB. For example blue color has Y component of 0.0722 for ITU-Recommendation BT.709, when it is stored in a 10 bit uint value, it becomes 0.722 * 1024 = 739.328 = 739 (we have to drop the fractional part here)

The following code should work
```
let blueImage = MTIImage(color: MTIColor(red: 0, green: 0, blue: 1, alpha: 1), sRGB: false, size: CGSize(width: 4, height: 4))
var pixelBuffer: CVPixelBuffer!
CVPixelBufferCreate(kCFAllocatorDefault, 4, 4, kCVPixelFormatType_420YpCbCr10BiPlanarFullRange, [kCVPixelBufferIOSurfacePropertiesKey as String: [:]] as CFDictionary, &pixelBuffer)
CVBufferSetAttachment(pixelBuffer, kCVImageBufferCGColorSpaceKey, CGColorSpace(name: CGColorSpace.sRGB)!, .shouldPropagate)
let context = try! MTIContext(device: MTLCreateSystemDefaultDevice()!)
try! context.render(blueImage, to: pixelBuffer)
var cgImage: CGImage!
VTCreateCGImageFromCVPixelBuffer(pixelBuffer, options: nil, imageOut: &cgImage)
var pixels: [UInt8] = [UInt8](repeating: 0, count: 4 * 4 * 4)
let cgContext = CGContext(data: &pixels, width: 4, height: 4, bitsPerComponent: 8, bytesPerRow: 4 * 4, space: CGColorSpace(name: CGColorSpace.sRGB)!, bitmapInfo: CGBitmapInfo.byteOrder32Little.rawValue | CGImageAlphaInfo.premultipliedFirst.rawValue)
cgContext?.draw(cgImage, in: CGRect(x: 0, y: 0, width: 4, height: 4))
for i in 0..<pixels.count {
    if i % 4 == 0 {
        assert(pixels[i] == 254) //b
        assert(pixels[i + 1] == 0) //g
        assert(pixels[i + 2] == 0) //r
        assert(pixels[i + 3] == 255) //a
    }
}
print("Passed")
```

Reply:
@YuAo Thanks, now it passes! :) I've updated the PR. Should we add it as an automatic test case, or these features are not supported in the simulators?

Reply:
Also feel free to tag me anytime you need further testing on tvOS!

Reply:
These features are not supported in the simulators, at least on Intel Macs. I highly doubt that these are supported on Apple Silicon Macs however I haven't yet test that.

Reply:
I think so too. But anyway, I'll get a M1 Air next week and I'm happy to do some manual testing there too! :D